src/App.tsx

```diff
@@ -1,24 +1,244 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import { nanoid } from 'nanoid';
+import React, {
+  useEffect, useState, useMemo, useRef,
+} from 'react';
+import { Todo, SortType } from './Types';
+import { UserWarning } from './components/UserWarning';
+import { getTodos } from './todos';
+import { client } from './utils/client';
+import { ErrorMessages } from './components/ErrorMessages';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
 
-const USER_ID = 0;
+const USER_ID = 10377;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTab, setSelectedTab] = useState(SortType.All);
+  const [inputValue, setInputValue] = useState('');
+  const [apiResponseReceived, setApiResponseReceived] = useState(true);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isThereActiveTodo, setIsThereActiveTodo] = useState(false);
+  const [isThereCompletedTodos, setIsThereCompletedTodos] = useState(false);
+  const [isUpdating, setIsUpdating] = useState(true);
+  const [isHidden, setIsHidden] = useState('');
+  const [placeHolderText, setPlaceHolderText] = useState('');
+  const [isDoubleClickedName, setIsDoubleClickedName] = useState('');
+  const [isThereIssue, setIsThereIssue] = useState(false);
+  const [numberOfActiveTodos, setNumberOfActivTodos] = useState(0);
+  const [deleteErrorMessage, setDeleteErrorMessage] = useState('');
+  const [isTitleEmpty, setIsTitleEmpty] = useState('');
+  const [updatingTodoId, setUpdatingTodoId] = useState<number | null>(null);
+  const [editTodo, setEditTodo] = useState('');
+  const [isEveryThingDelete, setIsEverythingDeleted] = useState(false);
+  const [deletedTodoId, setDeletedTodoId] = useState(0);
+  const [isPlusOne, setIsPlusOne] = useState(false);
+  const [isEveryThingTrue, setIsEveryThingTrue] = useState(false);
+  const [toggleFalseTodosId, setToggleFalseTodosId] = useState([0]);
+  const [todoStatusChange, setTodoStatusChange] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const excludedInputRef = useRef<HTMLInputElement>(null);
```

Comment:
Are you sure that you need to save all these states inside the App component? Would be better to move some of them to the components in which they are used